search/includes/classes/class-health.php

```diff
@@ -602,9 +573,12 @@ public function remove_validate_content_lock() {
 		delete_transient( self::CONTENT_VALIDATION_LOCK_NAME );
 	}
 
-	public function validate_index_posts_content_batch( $indexable, $start_post_id, $next_batch_post_id, $inspect ) {
+	public function validate_index_posts_content_batch( $indexable, $start_post_id, $next_batch_post_id, $options ) {
 		global $wpdb;
 
+		$inspect = isset( $options['inspect'] );
+		$mode    = $options['mode'] ?? null;
```

Comment:
What do you think about making the validation a bit more explicit, like

```
$supported_modes = [
	'complete',
	'missing',
	'mismatch', 
];

$mode    = isset( $options['mode'] ) && in_array( $options['mode'], $supported_modes, true ) ? $options['mode'] : 'complete';
```

This would also allow simplifying the `$mode` checks further down the method.

Reply:
Is it necessary if the CLI already throws an error when you attempt to pass in values that is not `mismatch` or `missing`? Also, unsure if we need a `complete` mode since it's there to limit a subset of index inconsistencies.

Reply:
Good points